org.eclipse.scanning.api/src/org/eclipse/scanning/api/device/IAttributableDevice.java

```diff
@@ -31,19 +31,19 @@
 	 * @throws ScanningException if the attribute cannot be retrieved for any reason
 	 */
 	public <T> IDeviceAttribute<T> getAttribute(String attributeName) throws ScanningException;
-	
+
 	/**
 	 * Gets a list of all attributes on the device.
-	 * @return all attributes
-	 * @throws ScanningException if the attributes cannot be retrieved for any reason 
+	 * @return all attributes, never <code>null</code>
+	 * @throws ScanningException if the attributes cannot be retrieved for any reason
 	 */
 	public List<IDeviceAttribute<?>> getAllAttributes() throws ScanningException;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1452)